services/transactionService.ts

```diff
@@ -72,45 +73,43 @@ export async function base64HashToHex (base64Hash: string): Promise<string> {
   )
 }
 
-export async function upsertTransaction (transaction: Prisma.TransactionUncheckedCreateInput, address: Address): Promise<TransactionWithAddressAndPrices | undefined> {
-  if (transaction.amount === new Prisma.Decimal(0)) { // out transactions
+export async function createTransaction (transactionData: Prisma.TransactionUncheckedCreateInput): Promise<TransactionWithAddressAndPrices | undefined> {
+  if (transactionData.amount === new Prisma.Decimal(0)) { // out transactions
     return
   }
-  const hash = await base64HashToHex(transaction.hash)
-  const transactionParams = {
-    hash,
-    amount: transaction.amount,
-    addressId: address.id,
-    timestamp: transaction.timestamp,
-    confirmed: transaction.confirmed
+  transactionData.hash = await base64HashToHex(transactionData.hash)
```

Comment:
Actually, I think this shouldn't be here. @vorg-code do you know why this is needed? With this refactor this would have to be done seperately also in the `createManyTransactions` method, but I think this should be done only when it's necessary, in the place that calls it. I think it's chronik, so I put it there, but I am not so sure